storage/backup/backup-cli/src/coordinators/backup.rs

```diff
@@ -353,10 +360,55 @@ impl BackupCompactor {
             state_snapshot_file_compact_factor,
             transaction_file_compact_factor,
             concurrent_downloads,
+            expiration_duration,
         }
     }
 
-    pub async fn run(&self) -> Result<()> {
+    /// Update the existing mapping and return the files to be moved out of metadata folder
+    fn update_compaction_timestamps(
+        &self,
+        meta_view: &mut MetadataView,
+        files: Vec<FileHandle>,
+        new_files: HashSet<FileHandle>,
+    ) -> Result<Vec<FileHandle>> {
+        // Get the current timestamp
+        let now = duration_since_epoch().as_secs();
+        // Iterate the metadata_compaction_timestamps and remove the expired files
+        let mut expired_files: Vec<FileHandle> = Vec::new();
+        let mut to_save_files: HashMap<FileHandle, Option<u64>> = HashMap::new();
+        let compaction_timestamps = meta_view
+            .select_latest_compaction_timestamps()
+            .as_ref()
+            .map(|meta| meta.compaction_timestamps.clone())
+            .unwrap_or_default();
+        for file in files {
+            // exclude newly compacted files
+            if new_files.contains(&file) {
+                continue;
+            }
+            if let Some(timestamp) = compaction_timestamps.get(&file.to_string()) {
+                if let Some(time_value) = timestamp {
+                    // file is in metadata_compaction_timestamps and expired
+                    if now > (*time_value + self.expiration_duration) {
+                        expired_files.push(file);
+                    } else {
+                        to_save_files.insert(file.to_string(), *timestamp);
+                    }
+                } else {
+                    to_save_files.insert(file.to_string(), Some(now));
+                }
+            } else {
+                to_save_files.insert(file.to_string(), Some(now));
+            }
+        }
+        // update the metaview compaction timestamps
+        let compaction_meta =
+            CompactionTimestampsMeta::new(to_save_files, duration_since_epoch().as_secs());
+        meta_view.update_compaction_timestamps(compaction_meta);
```

Comment:
So the metadata view is readonly and never updated before this, and I think it's good to keep it that way -- I think we can:
1. read the latest compaction_timestamps on construction, and never touch it again.
2. remove `metaview.update_compaction_timestamps()`
3. optional: I think it's better that we don't write down `newfile: None` entries (so CompactionTimestampsMeta::compaction_timestamps is `HashMap<FileHandle, u64>`, instead of 'HashMap<Filehandle, Option<u64>`) in the metadata file, but to your argument, there's not too many files so it's probably fine.

Reply:
removed update function and update it on construction now. 
For the optional one, too lazy to do it.

Reply:
@msmouse please take a look when you have time